File: apps/api/src/main/java/com/yalabs/yahub/mapper/UserMapper.java
package com.yalabs.yahub.mapper;

import java.time.LocalDateTime;
import java.util.UUID;

import com.yalabs.yahub.dtos.requests.UserRequest;
import com.yalabs.yahub.dtos.responses.UserResponse;
import com.yalabs.yahub.model.User;

public class UserMapper {
    public static User toEntity(UserRequest request, String passwordHash) {
        return User.builder()
        .id(UUID.randomUUID())
        .name(request.name())
        .email(request.email())
        .passwordHash(passwordHash)
        .createdAt(LocalDateTime.now())
        .build();
    }

    public static UserResponse toResponse(User user) {
    return new UserResponse(
        user.getId(),
        user.getName(),
        user.getEmail()
    );
}
}

File: Back-end/src/YaHub.Application/Mappers/UserMapper.cs
using YaHub.Application.DTOs.User;
using YaHub.Domain.Users;

namespace YaHub.Application.Mappers;

public sealed class UserMapper
{
    public User ToEntity(UserRequest request, string passwordHash)
    {
        return User.Create(
            request.Username,
            request.Email,
            passwordHash);
    }

    public UserResponse ToResponse(User user)
    {
        return new UserResponse
        {
            Username = user.Name,
            Email = user.Email
        };
    }

    public List<UserResponse> ToResponseList (List<User> users)
    {
        if (users == null) return new List<UserResponse>();

        return users.Select(users => ToResponse(users)).ToList();
    }
}